File: Later Work/Algorithms Theory And Programming Practice/Code Practice/Program 2/LinkedList.java

//Author: Brandon Fowler
//EWUID: 00639348
//Class: CSCD320
//Quarter: Spring 2014
//Programming Assignment 2

import java.util.*;

//Basic linked List with Nodes, constructor, addlast(), and a sort method
//==========================================================================================================
public class LinkedList{
	public Node head;             //Start of linked list
   public int length;            //Amount of nodes in LinkedList
	
   //Basic default constructor
   //=======================================================================================================
	public LinkedList(){
		this.head = null;
      this.length = 0;
	}
	
   //Adds a given Node to the end of the LinkedList
   //======================================================================================================
	public void addLast(Node n){
		if(this.head == null){                                         //If LinkedList is empty
			this.head = n;                                              //New node is the Head                                             
		}
		else{                                                          //If list is not empty
			Node cur;
			for(cur = this.head; cur.next != null; cur = cur.next);     //Iterate to end of list
			cur.next = n;                                               //Add the Node
		}
      length++;                                                      //Increase Node count
	}
   
   //Sorts a the LinkedList in ascending order based on vertex number
   //=====================================================================================================
   public void sort(){
      if(this.head != null && this.length > 1){           //If LinkedList is not empty or containing only one node
         
         //Preform basic sorting proceedure for LinkedList
         Node start = this.head;
         Node cur;
         while(start.next != null){
            cur = start;
            while(cur.next != null){
               if(cur.next.vertex < start.vertex){       //If smaller vertex is found
                  
                  //Swap vertex .d and path(required data for pinting)
                  int temp = start.vertex;
                  start.vertex = cur.next.vertex;
                  cur.next.vertex = temp;
                  
                  temp = start.d;
                  start.d = cur.next.d;
                  cur.next.d = temp;
                  
                  String t = start.path;
                  start.path = cur.next.path;
                  cur.next.path = t;  
               }
               cur = cur.next;
            }
            start = start.next;
         }
      }
   }
}

File: Later Work/C&Unix/fowlerblab4/linkedList.c
//CSCD340
//Lab4
//Brandon Fowler

#include "linkedList.h"
#define MAX 100


LinkedList * linkedList(){
	LinkedList * list = (LinkedList*)calloc(1,sizeof(LinkedList));
	return list;
}
Node * buildNode(){
	Node * ptr = (Node*)calloc(1,sizeof(Node));
	return ptr;
}
void sort(LinkedList * theList){
	Node * start = theList->head;
	Node * next = start->next;
	char * temp;
	while(next != NULL){
		while(next != start ){
			if(strncmp((char*)next->data,(char*)start->data,100) <0 ){
				temp = start->data;
				start->data = next->data;
				next->data = temp;
			}
			start = start->next;
		}
		start = theList->head;
		next = next->next;
	}	
}

void clearList(LinkedList * theList){
	Node * cur = theList->head;
	Node * prev;
	while(cur->next != NULL)
	{
		prev = cur;
		cur = cur->next;
		free(prev->data);
		free(prev);
		prev = NULL;
	}
	free(cur->data);
	free(cur);
	cur = NULL;
	free(theList);
}

void addLast(LinkedList * theList, Node * nn){
	if(theList->head == NULL)
	{
		theList->head = nn;
	}
	else
	{
		Node * cur = theList->head;
		for(cur;cur->next != NULL;cur=cur->next);
		cur->next = nn;
	}
	theList->size = theList->size+1;
}
void addFirst(LinkedList * theList, Node * nn){
	nn->next = theList->head;
	theList->head = nn;
	theList->size = theList->size+1;
}


